pkg/bindings/connection.go

```diff
@@ -108,23 +111,76 @@ func NewConnectionWithIdentity(ctx context.Context, uri string, identity string,
 	var connection Connection
 	switch _url.Scheme {
 	case "ssh":
+		userinfo := _url.User
+		if _url.User == nil {
+			u, err := user.Current()
+			if err != nil {
+				return nil, err
+			}
+			userinfo = url.User(u.Username)
+		}
 		port := 22
 		if _url.Port() != "" {
 			port, err = strconv.Atoi(_url.Port())
 			if err != nil {
 				return nil, err
 			}
 		}
+		// ssh_config
+		alias := _url.Hostname()
+		logrus.Debugf("Alias: %s", alias)
+		cfg := ssh_config.DefaultUserSettings
+		if val := cfg.Get(alias, "User"); val != "" {
+			userinfo = url.User(val)
+			logrus.Debugf("User: %s", val)
+		}
+		if val := cfg.Get(alias, "Hostname"); val != "" {
+			uri = val
+			logrus.Debugf("Hostname: %s", val)
+		}
+		if val := cfg.Get(alias, "Port"); val != "" {
+			if val != ssh_config.Default("Port") {
+				port, err = strconv.Atoi(val)
+				if err != nil {
+					return nil, err
```

Comment:
Same here, wrapped error would be appreciated (though I think this indicates a bad SSH config, we can at least tell them that)

Reply:
It was very unlikely to happen, since `ssh_config` library does not return a value for such a configuration:

`Error: ssh_config: strconv.ParseUint: parsing "42163xyz": invalid syntax`

But if it happens anyway, it is now wrapped. If you tried the same thing for url.Port, Parse would complain:

`invalid port ":123xyz" after host`